homeassistant/helpers/entityfilter.py

```diff
@@ -30,31 +39,110 @@ def convert_filter(config: Dict[str, List[str]]) -> Callable[[str], bool]:
         vol.Optional(CONF_EXCLUDE_DOMAINS, default=[]): vol.All(
             cv.ensure_list, [cv.string]
         ),
+        vol.Optional(CONF_EXCLUDE_ENTITY_GLOBS, default=[]): vol.All(
+            cv.ensure_list, [cv.string]
+        ),
         vol.Optional(CONF_EXCLUDE_ENTITIES, default=[]): cv.entity_ids,
         vol.Optional(CONF_INCLUDE_DOMAINS, default=[]): vol.All(
             cv.ensure_list, [cv.string]
         ),
+        vol.Optional(CONF_INCLUDE_ENTITY_GLOBS, default=[]): vol.All(
+            cv.ensure_list, [cv.string]
+        ),
         vol.Optional(CONF_INCLUDE_ENTITIES, default=[]): cv.entity_ids,
     }
 )
 
 FILTER_SCHEMA = vol.All(BASE_FILTER_SCHEMA, convert_filter)
 
 
+def convert_filter_alt(
+    config: Dict[str, Dict[str, List[str]]]
+) -> Callable[[str], bool]:
+    """Convert the alternate filter schema into a filter."""
```

Comment:
`alternate` doesn't tell the developer what this does.  We should name this something that shows intent

Reply:
Hm maybe `separated_schema`? Since this is for when `include` and `exclude` are specified separately as dictionaries?

Reply:
That still doesn't feel right. Might need to thinking about this one for a bit.  Naming is hard

Reply:
Maybe instead of trying to find a name that explains the schema should just call it `recorder` or `logbook` style. Those both use that schema and nearly everyone is familiar with them since they are key parts of core. That might resonate better then any descriptive name we could come up with.

Also I've seen the doc refer to the integrations referenced in this PR as recorder-like. Unfortunately that relationship is only subjective, there's no common recorder platform that all of them extend off of. But still I think using it here wouldn't be totally strange.

Reply:
Suggested names:
`convert_include_exclude_filter`
`convert_separated_filter`
`convert_bifurcated_filter`

Also it would be nice to have it wrap `convert_filter` instead of reimplementing so of the attribute setup.

Reply:
Lets go with `convert_include_exclude_filter`, feel like it most accurately conveys it.

> Also it would be nice to have it wrap convert_filter instead of reimplementing some of the attribute setup.

I can do that but I will still need to do some work in this method. After the method is complete the original config has be returned in the `config` attribute. So I will need to make a new object to call `convert_filter` against. Then replace the `config` attribute of the returned object with the original config.